flink-ml-lib/src/main/java/org/apache/flink/ml/feature/vectorassembler/VectorAssembler.java

```diff
@@ -47,10 +47,15 @@
 
 /**
  * A Transformer which combines a given list of input columns into a vector column. Types of input
- * columns must be either vector or numerical value.
+ * columns must be either vector or numerical types. The elements assembled in the same column must
+ * have the same size. The operator deals with null values or records with wrong sizes according to
+ * the strategy specified by the {@link HasHandleInvalid} parameter as follows:
  *
- * <p>The `keep` option of {@link HasHandleInvalid} means that we output bad rows with output column
- * set to null.
+ * <p>The `keep` option means that we do the assembling action without checking the vector size.
```

Comment:
Could you please add a detailed description of the behavior when "keep" option is set? As I can think of, any of the following might be a correct behavior.
- assembling `[1,2]`, `[3,4]` with sizes `2,3` may get `[1, 2, 3, 4]`
- assembling `[1,2]`, `[3,4]` with sizes `2,3` may get `[1, 2, 3, 4, 0]` (padding with zeros)
- assembling `[1,2]`, `[3,4,5]` with sizes `2,2` may get `[1, 2, 3, 4]` (trim to fit size)
- assembling `[1,2]`, `[3,4,5]` with sizes `2,2` may get `[1, 2, NaN, NaN]`
- assembling `[1,2]`, `null` with sizes `2,2` may get `[1, 2, NaN, NaN]`
- assembling `[1,2]`, `null` with sizes `2,2` may get `[1, 2, 0, 0]`

Let's check the option selected by Spark, implement that behavior, and clarify this behavior in JavaDoc.

Reply:
assembling [1,2], [3,4,5] with sizes 2,2 may get [1, 2, 3, 4] (trim to fit size)
assembling [1,2], [3,4,5] with sizes 2,2 may get [1, 2, NaN, NaN]
assembling [1,2], null with sizes 2,2 may get [1, 2, NaN, NaN]
assembling [1,2], null with sizes 2,2 may get [1, 2, 0, 0]